src/llvm/analysis/ReachingDefinitions/ReachingDefinitions.cpp

```diff
@@ -401,45 +400,104 @@ static bool isRelevantCall(const llvm::Instruction *Inst)
     assert(0 && "We should not reach this");
 }
 
-// return first and last nodes of the block
-std::pair<RDNode *, RDNode *>
+/**
+ * returns all RDBlock-s to which the LLVM block maps.
+ */
+std::vector<RDBlock *>
 LLVMRDBuilder::buildBlock(const llvm::BasicBlock& block)
 {
     using namespace llvm;
 
+    std::vector<RDBlock *> result;
+    RDBlock *rb = new RDBlock();
+    rb->setKey(const_cast<llvm::BasicBlock *>(&block));
+    addBlock(&block, rb);
+    result.push_back(rb);
+
     // the first node is dummy and serves as a phi from previous
     // blocks so that we can have proper mapping
     RDNode *node = new RDNode(RDNodeType::PHI);
     RDNode *last_node = node;
 
     addNode(node);
-    std::pair<RDNode *, RDNode *> ret(node, nullptr);
+    rb->append(node);
 
     for (const Instruction& Inst : block) {
         node = getNode(&Inst);
         if (!node) {
-           switch(Inst.getOpcode()) {
+            RDBlock *new_block = nullptr;
+            switch(Inst.getOpcode()) {
                 case Instruction::Alloca:
                     // we need alloca's as target to DefSites
-                    node = createAlloc(&Inst);
+                    node = createAlloc(&Inst, rb);
                     break;
                 case Instruction::Store:
-                    node = createStore(&Inst);
+                    node = createStore(&Inst, rb);
                     break;
                 case Instruction::Ret:
                     // we need create returns, since
                     // these modify CFG and thus data-flow
                     // FIXME: add new type of node NOOP,
                     // and optimize it away later
-                    node = createReturn(&Inst);
+                    node = createReturn(&Inst, rb);
+                    break;
+                case Instruction::GetElementPtr:
```

Comment:
Is the GetElementPtr correct? It does not read from memory.

Reply:
You're right, GetElementPtr doesn't need to be handled by RDA. 
fixed in 9f0c9f6